### DISCUSSION TITLE
Getting response headers

### DISCUSSION THREAD

**Discussion by vjuux** [TanStack/query#2046] [Q&A] [answered]
Is there a neat way to get the response headers nicely? :3
We are using 'X-Next' response header for our pagination that is being sent as query parameter within subsequent request.

currently doing this and wondering if there is a nicer way of handling such cases within `useInfiniteQuery`

```
  const xnext = payload.headers.get('X-Next');
  const data = await payload.json();
```

is the `getNextPageParam` the best place of getting `X-Next` header?

**Comment by TkDodo** [comment, marked as answer]
I would do that in the `queryFn` itself. The `getNextPageParam` is not async, so you can't do async json parsing there

**Reply by vjuux** [reply]
How would I notify `hasNextPage` in this case? Isn't that coming from getNextPageParam?

**Reply by TkDodo** [reply]
I would put all the necessary information from the `headers` into the `data` directly inside the `queryFn`, so you will have access to it in `getNextPageParam`
